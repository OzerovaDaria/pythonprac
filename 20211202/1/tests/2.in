class C(metaclass=dump):
    def __init__(self, s, *, debug=True):
        self.s = s
        self.p = False
        self.debug = debug

    def __add__(self, other = None):
        self.p = True
        pass

    def __str__(self):
        if (self.debug): return f'<<<dumpdump>>> {self.s}'
        return str(self.s)

a, b = C(10), C([4, 2, 6, 9, 1, 3, 3, 7], debug=True)
print(a, b)
print(b, a)

