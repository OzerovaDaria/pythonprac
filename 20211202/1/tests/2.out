__init__: (10,), {}
__init__: ([4, 2, 6, 9, 1, 3, 3, 7],), {'debug': True}
__str__: (), {}
<<<dumpdump>>> 10 __str__: (), {}
<<<dumpdump>>> [4, 2, 6, 9, 1, 3, 3, 7]
__str__: (), {}
<<<dumpdump>>> [4, 2, 6, 9, 1, 3, 3, 7] __str__: (), {}
<<<dumpdump>>> 10
